### PR TITLE
nm, wired: Read mac-address from sysfs when missing from NM

### PR DESCRIPTION
**Pull request by EdDev**
In some device types (e.g. ovs-interface type) NM does not provide the mac
address from the kernel.
For such cases, use sysfs as a fallback method to read the data.

### CODE REVIEW

File: libnmstate/nm/wired.py [hunk 1]
```diff
@@ -141,6 +143,17 @@ def get_info(device):
     return info
 
 
+def _get_mac_address_from_sysfs(ifname):
+    mac = None
+    sysfs_path = f'/sys/class/net/{ifname}/address'
+    try:
+        with open(sysfs_path) as f:
+            mac = f.read().rstrip('\n')
```
Comment (ffmancera) [inline libnmstate/nm/wired.py]: Black is complaining about the usage of `'`.

File: libnmstate/nm/wired.py [hunk 2]
```diff
@@ -130,6 +130,8 @@ def get_info(device):
 
     mac = device.get_hw_address()
     # A device may not have a MAC or it may not yet be "realized" (zeroed mac).
+    if not mac:
```
Comment (tyll) [inline libnmstate/nm/wired.py]: This needs to be inserted before the comment.

Comment (tyll) [inline libnmstate/nm/wired.py]: This needs an integration test to get the MAC address from a OVS interface.

File: libnmstate/nm/wired.py [hunk 3]
```diff
@@ -141,6 +143,17 @@ def get_info(device):
     return info
 
 
+def _get_mac_address_from_sysfs(ifname):
```
Comment (tyll) [inline libnmstate/nm/wired.py]: IMHO there should be a bug for NetworkManager to always provide the MAC and the bug should be referenced here in a docstring.

Comment (EdDev) [reply libnmstate/nm/wired.py]: Done

**Pull request merged**